api/solarperformanceinsight_api/models.py

```diff
@@ -864,12 +868,31 @@ class Job(BaseModel):
     system_definition: PVSystem
     parameters: JobParameters
     _data_items: List[JobDataItem] = PrivateAttr()
+    _weather_columns: List[str] = PrivateAttr()
+    _performance_columns: List[str] = PrivateAttr(["time", "performance"])
 
     def __init__(self, **data):
         super().__init__(**data)
         self._data_items = _construct_data_items(
             self.system_definition, self.parameters
         )
+        # determine what columns to expect in uploads
+        cols = [
+            "time",
+        ]
+        if self.parameters.irradiance_type == IrradianceTypeEnum.effective:
+            cols += ["effective_irradiance"]
+        elif self.parameters.irradiance_type == IrradianceTypeEnum.poa:
+            cols += ["poa_global", "poa_direct", "poa_diffuse"]
+        else:
+            cols += ["ghi", "dni", "dhi"]
+        if self.parameters.temperature_type == TemperatureTypeEnum.module:
+            cols += ["module_temperature"]
+        elif self.parameters.temperature_type == TemperatureTypeEnum.cell:
+            cols += ["cell_temperature"]
```

Comment:
I'd swap the order so that it's consistent with the irradiance enum ordering of least processed to most processed.